tests/unittest/llmapi/apps/test_tool_parsers.py

```
@@ -1504,6 +1505,60 @@ def test_encode_messages_multi_turn_with_tool_calls(self):
+
+
+class TestDeepSeekV4Parser(BaseToolParserTestClass):
+    """Test suite for DeepSeekV4Parser class."""
+
+    def make_parser(self):
+        return DeepSeekV4Parser()
+
+    def make_tool_parser_test_cases(self):
+        return ToolParserTestCases(
+            has_tool_call_true=
+            ('Some text <｜DSML｜tool_calls> <｜DSML｜invoke name="get_weather"> '
```

Ruff (0.15.18)

[warning] String contains ambiguous `｜` (FULLWIDTH VERTICAL LINE). Did you mean `|` (VERTICAL LINE)?

(RUF001)